### ISSUE TITLE
Different build environments.

### ISSUE THREAD

**Issue by man4ik** [zsoltk/paperwork#35]
Please provide a sample which has different values for different build variants.

**Comment by zsoltk** [comment]
I'm not sure I get what you mean. Please elaborate.

**Comment by man4ik** [comment]
```
 productFlavors {
    dev {
        buildConfigField 'String', 'EXAMPLE_0', '"some value 000"'
        buildConfigField 'boolean', 'EXAMPLE_1', 'true'
    }
    prod {
        buildConfigField 'String', 'EXAMPLE_0', 'some value 111'
        buildConfigField 'boolean', 'EXAMPLE_1', '"false"'
    }
```

How can I replace above code by paperwork?
Different values of same variables for different productFlavors

**Comment by zsoltk** [comment]
Right. I'm not sure if it's even theoretically possible to do that automatically, as you define your flavors / buildtypes in the `android` block. Maybe there's a workaround to access the current one outside of that scope, and someone will correct me on this.

What you can do however, is you could have a plugin config like:

``` groovy
paperwork {
    set = [
        debugKey1: "debug value",
        prodKey1: "prod value",
    ]
}
```

And then at runtime access it:

``` java
paperwork.get(BuildConfig.BUILD_TYPE + "Key1");
```

The plugin config supports arbitrary embedded structures, so in theory you could group all your debug keys in another set on level deeper without resorting to string concatenation. On the other hand, the runtime component only supports accessing a flat structure as of the moment.

**Comment by lordcodes** [comment]
I tried putting:

```
paperwork {
  set = [
     someKey = "some value"
  ]
}
```

Inside the debug block and then again in the release block to have a switch based on build type. Unfortunately, doing it this way means you have to duplicate all the keys that you want for both build types, as each paperwork block overwrites the previous.

I cannot use the prefix approach due to having another build type for development environment and would also rather not use a prefix-based approach.

Would it be possible to make it so that you can have multiple different paperwork blocks and have all the variables be added to the JSON file?

**Comment by zsoltk** [comment]
@andrewlord1990 I think what you want to achieve has to do more with groovy than with paperwork, i.e. merging those arrays. Also, why not try the same key approach, but instead of literals, you could have function calls on the right side, then override those functions per environment. I haven't checked, but it would make sense if it worked.

**Comment by lordcodes** [comment]
Okay thanks, interesting idea. Will try that out when I get a chance.
